model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -317,7 +318,9 @@ public boolean removeIdentityProvider(OrganizationModel organization, IdentityPr
             return false;
         }
 
+        // clear the organization id and any domain assigned to the IDP.
         identityProvider.setOrganizationId(null);
+        identityProvider.getConfig().remove(ORGANIZATION_DOMAIN_ATTRIBUTE);
```

Comment:
Another org-related attribute we set is https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/OrganizationModel.java#L29.

We should also remove it ...

Reply:
Fixed!